agent/fw_wordpress.c

```diff
@@ -551,6 +553,65 @@ NR_PHP_WRAPPER(nr_wordpress_apply_filters) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
+/*
+ * Wrap the wordpress function add_filter
+ *
+ * function add_filter( $hook_name, $callback, $priority = 10, $accepted_args = 1 )
+ *
+ * @param string   $hook_name     The name of the filter to add the callback to.
+ * @param callable $callback      The callback to be run when the filter is applied.
+ * @param int      $priority      Optional. Used to specify the order in which the functions
+ *                                associated with a particular filter are executed.
+ *                                Lower numbers correspond with earlier execution,
+ *                                and functions with the same priority are executed
+ *                                in the order in which they were added to the filter. Default 10.
+ * @param int      $accepted_args Optional. The number of arguments the function accepts. Default 1.
+ * @return true Always returns true.
+ */
+
+NR_PHP_WRAPPER(nr_wordpress_add_filter) {
+  /* Wordpress's add_action() is just a wrapper around add_filter(),
+   * so we only need to instrument this function */
+  NR_UNUSED_SPECIALFN;
+  (void)wraprec;
+  const char* filename = NULL;
+  bool wrap_hook = true;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_WORDPRESS);
+
+  /*
+   * We only want to hook the function being called if this is a WordPress
+   * function, we're instrumenting hooks, and WordPress is currently executing
+   * hooks (denoted by the wordpress_tag being set).
+   */
+  if ((NR_FW_WORDPRESS != NRPRG(current_framework))
+      || (0 == NRINI(wordpress_hooks))) {
+    wrap_hook = false;
+  }
+
+  if (NRINI(wordpress_hooks_skip_filename)
+      && 0 != nr_strlen(NRINI(wordpress_hooks_skip_filename))) {
```

Comment:
Would it be faster to just check if the string `NRINI(wordpress_hooks_skip_filename)` doesn't start with a byte equal to `0`?

Reply:
@mfulb I think it would even be better to replace these two checks (`NULL` and `nr_strlen`) with a call to `nr_strempty` which does just what is needed here. I would however prefer to do it in a separate PR as this code needs cleanup anyway because it is repeated twice: [here](https://github.com/newrelic/newrelic-php-agent/blob/5a1ef2bcd7ed1e5ffde87c9175cce8e41abb8268/agent/fw_wordpress.c#L344-L353) and [here](https://github.com/newrelic/newrelic-php-agent/blob/5a1ef2bcd7ed1e5ffde87c9175cce8e41abb8268/agent/fw_wordpress.c#L593-L602).